jni/cmake/init-faiss.cmake

```diff
@@ -54,18 +54,37 @@ if(NOT DEFINED APPLY_LIB_PATCHES OR "${APPLY_LIB_PATCHES}" STREQUAL true)
     endforeach()
 endif()
 
-if (${CMAKE_SYSTEM_NAME} STREQUAL Darwin)
-    if(CMAKE_C_COMPILER_ID MATCHES "Clang\$")
-        set(OpenMP_C_FLAGS "-Xpreprocessor -fopenmp")
-        set(OpenMP_C_LIB_NAMES "omp")
-        set(OpenMP_omp_LIBRARY /usr/local/opt/libomp/lib/libomp.dylib)
-    endif()
+if (APPLE)
+    message(STATUS "darwin macos detected")
+    if(CMAKE_SYSTEM_PROCESSOR STREQUAL "arm64")
+        message(STATUS "detected Mac with ARM architecture")
+        if(CMAKE_C_COMPILER_ID MATCHES "Clang\$")
+            set(OpenMP_C_FLAGS "-Xpreprocessor -fopenmp")
+            set(OpenMP_C_LIB_NAMES "omp")
+            set(OpenMP_omp_LIBRARY /opt/homebrew/opt/libomp/lib/libomp.dylib)
```

Comment:
Is this also going to be system dependent?

Reply:
yes this is system independent. Newer version of brew will use that path by default while the old ones use the old path that was hard coded.

Reply:
My concern is more that users may not use brew to install omp

Reply:
The guideline in the instructions right now for Mac is to use brew, if this guideline is followed the build will fail unless the above is set.